Sources/WalletConnectRelay/Dispatching.swift

```diff
@@ -69,13 +77,13 @@ final class Dispatcher: NSObject, Dispatching {
     }
 
     func protectedSend(_ string: String, completion: @escaping (Error?) -> Void) {
-        guard !socket.isConnected else {
+        guard !socket.isConnected || !networkMonitor.isConnected else {
             return send(string, completion: completion)
         }
 
         var cancellable: AnyCancellable?
-        cancellable = socketConnectionStatusPublisher
-            .filter { $0 == .connected }
+        cancellable = Publishers.CombineLatest(socketConnectionStatusPublisher, networkConnectionStatusPublisher)
+            .filter { $0.0 == .connected && $0.1 == .connected }
             .setFailureType(to: NetworkError.self)
             .timeout(.seconds(defaultTimeout), scheduler: concurrentQueue, customError: { .webSocketNotConnected })
```

Comment:
should we use different more generic error?

Reply:
changed to .connectionFailed